libusb/os/windows_winusb.c

```diff
@@ -1380,16 +1381,23 @@ static int set_composite_interface(struct libusb_context *ctx, struct libusb_dev
 	int interface_number;
 	const char *mi_str;
 	int iadi, iadintfi;
+	char* endptr;
 	struct libusb_interface_association_descriptor_array *iad_array;
 	const struct libusb_interface_association_descriptor *iad;
 
 	// Because MI_## are not necessarily in sequential order (some composite
 	// devices will have only MI_00 & MI_03 for instance), we retrieve the actual
 	// interface number from the path's MI value
 	mi_str = strstr(device_id, "MI_");
-	if ((mi_str != NULL) && isdigit((unsigned char)mi_str[3]) && isdigit((unsigned char)mi_str[4])) {
-		interface_number = ((mi_str[3] - '0') * 10) + (mi_str[4] - '0');
-	} else {
+
+	endptr = NULL;
+	interface_number = 0;
```

Comment:
Since this is only for the compiler and not for setting a default value, I think it would be best to initialize it to e.g. -1 here, just to make a point to the human reader.

Reply:
What point would that be?

Reply:
That this is an initialization value that won't be carried over, and is there only to make the compiler happy. Otherwise someone may smell redundancy and be tempted to remove it. I would have added a comment too.

Reply:
Fair enough. Although removing it will instantly result in a failed build, so it's not like it is a hidden bug. Changed/commented

Reply:
Thanks! Yes, the CI would catch it. But gcc wouldn't complain, so it might not be seen immediately by everyone.